docs/rst/tutorials/micro/user_api/user_api_middleware.rst

```diff
@@ -0,0 +1,279 @@
+
+An usage example can be found on `epoch_synchronization micro-ROS demo <https://github.com/micro-ROS/micro-ROS-demos/blob/humble/rclc/epoch_synchronization/main.c>`_:
+
+.. code-block:: c
+
+    // Sync timeout
+    const int timeout_ms = 1000;
+
+    // Synchronize time with the agent
+    rmw_uros_sync_session(timeout_ms);
+
+
+After the session is synchronized, the adjusted timestamp can be retrieved with the following API:
+
+.. code-block:: c
+
+    // Check if session has been synchronized
+    if (rmw_uros_epoch_synchronized())
+    {
+        // Get time in milliseconds or nanoseconds
+        int64_t time_ms = rmw_uros_epoch_millis();
+        int64_t time_ns = rmw_uros_epoch_nanos();
+    }
+
+.. note::
+
+    micro-ROS shall be already initialized and connected to the agent to use this functionality.
+
+Ping agent
+----------
+
+The Client can test the connection with the Agent with the ping utility. This functionality can be used even when the micro-ROS context has not yet been initialized, which is useful to test the connection before trying to connect to the Agent. An example can be found on `ping_uros_agent micro-ROS demo <https://github.com/micro-ROS/micro-ROS-demos/blob/humble/rclc/ping_uros_agent/main.c>`_.
+
+.. code-block:: c
+
+    // Timeout for each attempt
+    const int timeout_ms = 1000;
+
+    // Number of attemps
+    const uint8_t attemps = 5;
+
+    // Ping the agent
+    rmw_ret_t ping_result = rmw_uros_ping_agent(timeout_ms, attempts);
+
+    if (RMW_RET_OK == ping_result)
+    {
+        // micro-ROS Agent is reachable
+        ...
+    }
+    else
+    {
+        // micro-ROS Agent is not available
+        ...
+    }
+
+A secondary API is provided to ping the Agent with a specific rmw configuration. This API allows the user to ping with a specific custom transport without interfering with the actual micro-ROS configuration:
+
+.. code-block:: c
+
+    // Initialize rcl options and retrieve the internal rmw options
+    rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+    rcl_init_options_init(&init_options, allocator);
+    rmw_init_options_t * rmw_options = rcl_init_options_get_rmw_init_options(&init_options);
+
+    // Set custom transport
+    rmw_uros_options_set_custom_transport(
+        false,
+        NULL,
+        custom_transport_open,
+        custom_transport_close,
+        custom_transport_write,
+        custom_transport_read,
+        rmw_options);
+
+    // Ping the agent with custom transport
+    const int timeout_ms = 1000;
+    const uint8_t attemps = 5;
+    rmw_ret_t ping_result = rmw_uros_ping_agent_options(timeout_ms, attempts, rmw_options);
+
+    if (RMW_RET_OK == ping_result)
+    {
+        // micro-ROS Agent is reachable
+        ...
+    }
+    else
+    {
+        // micro-ROS Agent is not available
+        ...
+    }
+
+Continuous serialization
+------------------------
+
+..   TODO(acuadros95): Do we have an example of this?
```

Comment:
Not sure about posting this because it has some considerations on types that maybe need to explained detailed. Should we remove this?

Reply:
Agree, deleted here https://github.com/eProsima/vulcanexus/pull/78/commits/346a5be6837282f7b23d35cfe1375a8f7aa19ecd